doc/source/reference/c-api.types-and-structures.rst

```diff
@@ -537,13 +537,14 @@ PyArrayDescr_Type
         the array memory pointed to by ``ip``. The array is assumed
         to be behaved. If ``sep`` is not NULL, then a separator string
         is also scanned from the file before returning. The last
-        argument ``arr`` is the array to be scanned into. A 0 is
-        returned if the scan is successful. A negative number
-        indicates something went wrong: -1 means the end of file was
-        reached before the separator string could be scanned, -4 means
-        that the end of file was reached before the element could be
-        scanned, and -3 means that the element could not be
-        interpreted from the format string. Requires a behaved array.
+        argument ``arr`` is the array to be scanned into.
+        Returns number of receiving arguments successfully assigned (which
+        may be zero in case a matching failure occurred before the first
+        receiving argument was assigned), or EOF if input failure occurs 
+        before the first receiving argument was assigned.
+        Requires a behaved array.
+        This function should be called without holding the Python GIL, and
+        has to grab it for error reporting.
```

Comment:
"Requires a behaved array" is duplicated, see the unchanged text. Also, it seems the use of `sep` is deprecated, the only use of `scanfunc` in the code (via [`fromfile_next_element`](https://github.com/numpy/numpy/blob/v1.16.4/numpy/core/src/multiarray/ctors.c#L72)) will force it to `NULL`.

Reply:
@mattip Thanks, I've made the changes. Does it look ok?